cranelift/codegen/src/isa/s390x/inst/regs.rs

```diff
@@ -78,6 +78,10 @@ pub fn zero_reg() -> Reg {
     gpr(0)
 }
 
+pub fn link_reg() -> Reg {
```

Comment:
I don't really like this.  This function, with this name in this place, gives the wrong impression that this is a ISA-defined register, and encourages additional uses in the future.

If we need a function like this, it should be an *ABI* function.

In any case, if you really want to hard-code the link register in `ElfTlsGetOffset`, then I'd suggest to just hard-code 14 directly, like this code also hard-codes register numbers 12 and 2.  (Again, I'd prefer to go the other way and not hard-code any of them, and instead get them from ABI code.  But given that this opcode is ABI-specific, I don't really have a strong opinion here.)

Reply:
Inlined in all callers.

Reply:
LGTM now, thanks!